pkg/inventory/inventory-client.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package inventory
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/klog"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/validation"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+// InventoryClient expresses an interface for retrieving Inventory
+// objects in the clus
+type InventoryClient interface {
+	// GetPreviousInventoryObjects returns the inventory objects (as *resource.Info)
+	// that are currently stored in the cluster, or an error if one occurred. Uses
+	// information from the current (recently) created inventory object to find
+	// the previously applied inventory object in the cluster.
+	GetPreviousInventoryObjects(currentInv *resource.Info) ([]*resource.Info, error)
```

Comment:
Longer term I don't think we should have a resource.Info as the parameter here, but rather something like an `InventoryID` that just captures the info we need to look up the previous inventories.